### PR TITLE
chore: benchmark Protobuf-ES

### PR DESCRIPTION
**Pull request by felicio**
Repo: https://github.com/bufbuild/protobuf-es
Release: https://buf.build/blog/protobuf-es-the-protocol-buffers-typescript-javascript-runtime-we-all-deserve

---

Relates to https://github.com/bufbuild/protobuf-es/issues/333

### CODE REVIEW

File: packages/protons-benchmark/README.md [hunk 1]
```diff
@@ -16,20 +16,17 @@
...
 
-```console
-$ npm i protons-benchmark
-```
-
 ```console
 $ git clone git@github.com:ipfs/protons.git
 $ cd protons
 $ npm i
+$ npm run build
```
Comment (felicio) [inline packages/protons-benchmark/README.md:23]: ⚠️ Without this:

```
$ npm start                     

> protons-benchmark@0.0.0 prestart
> npm run build


> protons-benchmark@0.0.0 build
> aegir build --no-bundle && cp -R src/protobufjs dist/src/protobufjs

[22:08:47] tsc [started]
src/protons/bench.ts:6:68 - error TS2307: Cannot find module 'protons-runtime' or its corresponding type declarations.

6 import { encodeMessage, decodeMessage, message, enumeration } from 'protons-runtime'

...

Found 89 errors in 2 files.

Errors  Files
    32  src/protons/bench.ts:6
    57  src/protons/rpc.ts:6
[22:08:48] tsc [failed]
[22:08:48] → Command failed with exit code 1: tsc

...
```

File: packages/protons-benchmark/README.md [hunk 2]
```diff
@@ -47,10 +44,11 @@ $ npm start
...
 
-pbjs x 11,798 ops/sec ±4.58% (88 runs sampled)
-protons x 11,693 ops/sec ±2.69% (85 runs sampled)
-protobuf.js x 12,419 ops/sec ±1.66% (88 runs sampled)
-@protobuf-ts x 10,536 ops/sec ±3.14% (85 runs sampled)
+pbjs x 19,188 ops/sec ±0.38% (98 runs sampled)
+protons x 19,001 ops/sec ±0.33% (95 runs sampled)
+protobuf.js x 19,558 ops/sec ±0.30% (91 runs sampled)
+@protobuf-ts x 17,216 ops/sec ±0.32% (95 runs sampled)
+protobuf-es x 15,673 ops/sec ±0.48% (93 runs sampled)
```
Comment (felicio) [inline packages/protons-benchmark/README.md:51]: ℹ️ Slowest.

File: packages/protons-benchmark/README.md [hunk 3]
```diff
@@ -63,11 +61,12 @@ $ npm run start:browser
...
-pbjs x 19,027 ops/sec ±0.86% (67 runs sampled)
-protons x 18,901 ops/sec ±0.65% (67 runs sampled)
-protobuf.js x 18,937 ops/sec ±0.55% (65 runs sampled)
-@protobuf-ts x 16,669 ops/sec ±0.49% (68 runs sampled)
-Fastest is pbjs,protobuf.js
+pbjs x 19,763 ops/sec ±0.35% (67 runs sampled)
+protons x 19,617 ops/sec ±0.37% (68 runs sampled)
+protobuf.js x 19,772 ops/sec ±0.34% (67 runs sampled)
+@protobuf-ts x 17,204 ops/sec ±0.33% (69 runs sampled)
+protobuf-es x 16,032 ops/sec ±0.38% (68 runs sampled)
```
Comment (felicio) [inline packages/protons-benchmark/README.md:68]: ℹ️ Slowest.

**Review by achingbrain — CHANGES_REQUESTED** [review]
Thanks for opening this & sorry it's taken me so long to get to this PR!

LGTM - just a question mark over the two yaml files that have been added - what are they for?

File: packages/protons-benchmark/buf.gen.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+version: v1
```
Comment (achingbrain) [inline packages/protons-benchmark/buf.gen.yaml:1]: Is there a way for this config file to live in the `protobuf-es` folder?  It'd be more manageable not to have implementation-specific files outside of their respective directories.

File: packages/protons-benchmark/src/buf.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+version: v1
```
Comment (achingbrain) [inline packages/protons-benchmark/src/buf.yaml:1]: Same comment as https://github.com/ipfs/protons/pull/89/files#r1357807337

**Pull request merged**